pkg/plugins/resources/dockerimage/condition.go

```diff
@@ -16,44 +12,25 @@ import (
 // We assume that if we can't retrieve the docker image digest, then it means
 // it doesn't exist.
 func (di *DockerImage) Condition(source string, scm scm.ScmHandler, resultCondition *result.Condition) error {
-
 	if scm != nil {
 		logrus.Warningf("SCM configuration is not supported for condition of type dockerimage. Remove the `scm` directive from condition to remove this warning message")
 	}
 
-	refName := di.spec.Image
-	switch di.spec.Tag == "" {
-	case true:
-		refName += ":" + source
-	case false:
-		refName += ":" + di.spec.Tag
-	}
-
-	ref, err := name.ParseReference(refName)
+	ref, err := di.createRef(source)
 	if err != nil {
-		return fmt.Errorf("invalid image %s: %w", refName, err)
+		return err
 	}
 
-	found := false
-	switch len(di.spec.Architectures) {
-	case 0:
-		// If only one architecture is specified, then di.options are already set: let's proceed
-		found, err = checkImage(ref, di.spec.Architecture, di.options)
+	found := true
+
+	for _, arch := range di.spec.Architectures {
```

Comment:
This block has been changed how it worked previously, it will now fail if _any_ are not found.